app/plugins/adp-backend/src/service/deliveryProgrammeRouter.ts

```diff
@@ -0,0 +1,252 @@
+    const data = await deliveryProgrammesStore.getAll();
+    res.json(data);
+  });
+
+  router.get('/programmeManager', async (_req, res) => {
+    const data = await programmeManagersStore.getAll();
+    res.json(data);
+  });
+
+  router.get('/deliveryProgramme/:id', async (_req, res) => {
+    const deliveryProgramme = await deliveryProgrammesStore.get(_req.params.id);
+    const programmeManager = await programmeManagersStore.get(_req.params.id);
+    if (programmeManager && deliveryProgramme !== null) {
+      deliveryProgramme.programme_managers = programmeManager;
+      res.json(deliveryProgramme);
+    }
+  });
+
+  router.get('/catalogEntities', async (_req, res) => {
+    const catalogApiResponse = await catalog.getEntities({
+      filter: {
+        kind: 'User',
+      },
+      fields: [
+        'metadata.name',
+        'metadata.annotations.graph.microsoft.com/user-id',
+        'metadata.annotations.microsoft.com/email',
+      ],
+    });
+    res.json(catalogApiResponse);
+  });
+
+  router.post('/deliveryProgramme', async (req, res) => {
+    try {
+      if (!isDeliveryProgrammeCreateRequest(req.body)) {
+        throw new InputError('Invalid payload');
+      }
+
+      const data: DeliveryProgramme[] = await deliveryProgrammesStore.getAll();
+
+      const isDuplicate: boolean = await checkForDuplicateTitle(
+        data,
+        req.body.title,
+      );
+      if (isDuplicate) {
+        res
+          .status(406)
+          .json({ error: 'Delivery Programme title already exists' });
+      } else {
+        const author = await getCurrentUsername(identity, req);
+        const deliveryProgramme = await deliveryProgrammesStore.add(
+          req.body,
+          author,
+        );
+
+        const programmeManagers = req.body.programme_managers;
+        if (programmeManagers !== undefined) {
+          const catalogEntities = await catalog.getEntities({
+            filter: {
+              kind: 'User',
+            },
+            fields: [
+              'metadata.name',
+              'metadata.annotations.graph.microsoft.com/user-id',
+              'metadata.annotations.microsoft.com/email',
+            ],
+          });
+
+          const catalogEntity: Entity[] = catalogEntities.items;
+
+          addProgrammeManager(
+            programmeManagers,
+            deliveryProgramme.id,
+            deliveryProgramme,
+            programmeManagersStore,
+            catalogEntity,
+          );
+        } else {
+          req.body.programme_managers = [];
+        }
+        res.json(deliveryProgramme);
+      }
+    } catch (error) {
+      throw new InputError('Error');
+    }
+  });
+
+  router.patch('/deliveryProgramme', async (req, res) => {
+    try {
+      if (!isDeliveryProgrammeUpdateRequest(req.body)) {
+        throw new InputError('Invalid payload');
+      }
+      const data: DeliveryProgramme[] = await deliveryProgrammesStore.getAll();
```

Comment:
Getting all of the programmes is required for the checkForDuplicateTitle function. I've refactored it so it does getAll() and  a get() to fetch the current object instead of just doing getAll() and doing a find. Is this a better approach? @jamescoxheadcapgemini